dex/src/main/scala/com/wavesplatform/dex/api/routes/ApiRoute.scala

```diff
@@ -40,6 +41,22 @@ trait ApiRoute extends Directives with ApiMarshallers with ScorexLogging {
 
   protected val invalidUserPublicKey: StandardRoute = complete(SimpleErrorResponse(error.UserPublicKeyIsNotValid()))
 
+  protected def signedGetByAddress(addressOrError: Either[ValidationError.InvalidAddress, Address]): Directive0 =
+    addressOrError match {
+      case Left(ia) => complete(SimpleErrorResponse(error.InvalidAddress(ia.reason)))
+      case Right(address) =>
+        (headerValueByName(`X-User-Public-Key`.headerName) & headerValueByName("Timestamp") & headerValueByName("Signature")).tflatMap {
+          case (publicKey, timestamp, sig) =>
+            val pk = PublicKey.fromBase58String(publicKey).getOrElse(PublicKey.empty)
+            if (pk.toAddress != address) complete(InvalidSignature)
```

Comment:
Interesting. Could we reuse `signedGet`? We have this condition in `cancelOrdersByIdsWithKeyOrSignature`

Reply:
fixed